sdcm/cluster_gce.py

```diff
@@ -94,34 +95,34 @@ def tags(self) -> Dict[str, str]:
                 "NodeIndex": str(self.node_index), }
 
     def _set_keep_alive(self) -> bool:
-        return self._instance_wait_safe(self._gce_service.ex_set_node_labels, self._instance, {"keep": "alive"}) and \
+        return self._instance_wait_safe(self._instance.add_tags, {"keep": "alive"}) and \
             super()._set_keep_alive()
 
     def _instance_wait_safe(self, instance_method: Callable[P, R], *args: P.args, **kwargs: P.kwargs) -> R:
         try:
             return exponential_retry(func=lambda: instance_method(*args, **kwargs), logger=self.log)
         except tenacity.RetryError:
             raise cluster.NodeError(
-                f"Timeout while running '{instance_method.__name__}' method on GCE instance '{self._instance.id}'"
+                f"Timeout while running '{instance_method.__name__}' method on GCE instance '{self._instance.name}'"
             ) from None
 
     def _refresh_instance_state(self):
         node_name = self._instance.name
-        instance = self._instance_wait_safe(self._gce_service.ex_get_node, node_name)
-        self._instance = instance
-        ip_tuple = (instance.public_ips, instance.private_ips)
-        return ip_tuple
+        node = self._instance_wait_safe(self._gce_service.ex_get_node, node_name)
```

Comment:
@KnifeyMoloko do you know what is reason of existence of `_refresh_instance_state`? For me it looks that it updates ip addresses - do they change?

Reply:
I don't know the reason why it was introduced.